pkg/api/validation/validation.go

```diff
@@ -1653,8 +1654,9 @@ func validateContainerResourceDivisor(rName string, divisor resource.Quantity, f
 func validateConfigMapKeySelector(s *api.ConfigMapKeySelector, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if len(s.Name) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath.Child("name"), ""))
+	nameFn := apimachineryvalidation.ValidateNameFunc(ValidateSecretName)
```

Comment:
How about use `ValidateNameFunc` directly here? As we already have the `ValidateNameFunc` here: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L228